src/Components/Register/Register.jsx

```diff
@@ -19,42 +19,47 @@ export default class Register extends Component {
 
   handleSubmit(event) {
     event.preventDefault();
+    const { dispatch, firstName, lastName, email, password } = this.props;
 
-    if (this.props.confirmPassword === this.props.password) {
-      const { dispatch, firstName, lastName, email, password } = this.props;
-      dispatch(register({ firstName, lastName, email, password }, dispatch ))
+    if (!this.props.email.includes('.', this.props.email.indexOf('@') + 2) || this.props.email.substring(this.props.email.lastIndexOf('.')).length <= 2) {
+      alert('Email is not valid.');
     } else {
-      alert('Passwords do not match');
+      if (this.props.confirmPassword === this.props.password) {
```

Comment:
Destructure `confirmPassword` on line 22 and remove `this.props` from both password and confirmPassword

Reply:
Resolved.